### ISSUE TITLE
Why TransformerDecoder has multiple LayerNorm and Dropout initial 

### ISSUE THREAD

**Issue by XD666** [facebookresearch/detr#464]
Any reseason for using so many LN and dropout?

 self.norm1 = nn.LayerNorm(d_model)
        self.norm2 = nn.LayerNorm(d_model)
        self.norm3 = nn.LayerNorm(d_model)
        self.dropout1 = nn.Dropout(dropout)
        self.dropout2 = nn.Dropout(dropout)
        self.dropout3 = nn.Dropout(dropout)

Why not just one LayerNorm and Dropout to reduce the memory? as following? 

 self.norm1 = nn.LayerNorm(d_model)
 self.dropout1 = nn.Dropout(dropout)
